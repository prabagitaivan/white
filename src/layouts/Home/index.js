import React, { PureComponent } from 'react'
import Navigator from '../../components/Navigator'
import Content from '../../components/Content'

export default class Home extends PureComponent {
  render () {
    return (
      <div>
        <Navigator />
        <Content>Home</Content>
      </div>
    )
  }
}
