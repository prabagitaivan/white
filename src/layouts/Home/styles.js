import { makeStyles } from '@material-ui/styles'

const layout = 'Home'

export const loading = makeStyles(
  {
    root: {
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center'
    },
    linearProgress: {
      height: 10,
      width: '60%'
    }
  },
  { name: `${layout}Loading` }
)

export const empty = makeStyles(
  {
    root: {
      textAlign: 'center'
    }
  },
  { name: `${layout}Empty` }
)

export const notes = makeStyles(
  {
    img: {
      cursor: 'pointer'
    },
    gridListTileBar: ({ desktop }) => ({
      height: desktop ? 48 : 40
    }),
    title: {
      fontSize: ({ desktop }) => (desktop ? 24 : 15),
      lineHeight: 'normal',
      whiteSpace: 'normal',
      color: 'rgba(255, 255, 255, 0.85)'
    }
  },
  { name: `${layout}Notes` }
)
