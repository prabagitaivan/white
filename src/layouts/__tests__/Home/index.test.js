import { render } from '@testing-library/react'
import React from 'react'
import Stores from '../../../stores'
import Styles from '../../../styles'
import Home from '../../Home'

function renderHome (desktop = true) {
  const preloadedState = { status: { desktop } }

  return render(
    <Stores preloadedState={preloadedState}>
      <Styles>
        <Home />
      </Styles>
    </Stores>
  )
}

describe('Home', () => {
  describe('snapshots', () => {
    it('contain correct content', () => {
      const { container } = renderHome()
      expect(container.querySelector('#content').firstChild).toMatchSnapshot()
    })
  })
})
