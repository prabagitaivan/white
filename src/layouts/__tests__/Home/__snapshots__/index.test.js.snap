// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshots contain correct content 1`] = `Home`;
